riscv/tests/common/mod.rs

```diff
@@ -28,25 +28,36 @@ pub fn verify_riscv_asm_string<T: FieldElement, S: serde::Serialize + Send + Syn
         pipeline = pipeline.add_data_vec(data);
     }
 
+    // verify with PILCOM
+    if T::known_field().unwrap() == KnownField::GoldilocksField {
+        let pipeline_gl: Pipeline<GoldilocksField> =
+            unsafe { std::mem::transmute(pipeline.clone()) };
+        run_pilcom_with_backend_variant(pipeline_gl, BackendVariant::Composite).unwrap();
+    }
+
+    test_plonky3_pipeline::<T>(pipeline.clone());
+
+    // verify using executor generated witness
     // TODO remove the guard once the executor is implemented for BB
     if T::known_field().unwrap() == KnownField::GoldilocksField {
         let analyzed = pipeline.compute_analyzed_asm().unwrap().clone();
-        powdr_riscv_executor::execute_ast(
+        let pil = pipeline.compute_optimized_pil().unwrap();
+        let fixed = pipeline.compute_fixed_cols().unwrap().clone();
+        let execution = powdr_riscv_executor::execute(
             &analyzed,
+            &pil,
+            fixed,
             Default::default(),
             pipeline.data_callback().unwrap(),
-            // Assume the RISC-V program was compiled without a bootloader, otherwise this will fail.
             &[],
-            usize::MAX,
-            powdr_riscv_executor::ExecMode::Fast,
-            Default::default(),
+            None,
+            None,
         );
-        let pipeline_gl: Pipeline<GoldilocksField> =
-            unsafe { std::mem::transmute(pipeline.clone()) };
-        run_pilcom_with_backend_variant(pipeline_gl, BackendVariant::Composite).unwrap();
+        pipeline.rollback_from_witness();
+        let executor_trace: Vec<_> = execution.trace.into_iter().collect();
+        let pipeline = pipeline.add_external_witness_values(executor_trace);
```

Comment:
Should we add a check somewhere in the tests that all cols were computed? Or does that not make sense atm?

Reply:
idk if it makes sense to break tests for that rn, as we're not using it yet. It will only break if proving fails or if its partial and witgen can't finish it.

Reply:
Ok. At some point (next PRs I guess) it would be good to get quick feedback from CI on whether all columns are there. Maybe some specific tests.